src/Utilities/InternetParser.cs

```diff
@@ -44,18 +43,31 @@ public static async Task ReadPriceFromCamel(string itemName)
                     .Equals("green"))
                     ?.InnerText;
             stringPrice = stringPrice?.Substring(1).Trim();
-
+            if (stringPrice == null)
+            {
+                return Goal.CreateLocalGoal(itemName, 0);
+            }
 
             try
             {
-                await File.WriteAllTextAsync(ResourceDirectoryParsedGoal, name + "\n" + stringPrice);
+                var decimalPrice = Convert.ToDecimal(stringPrice, System.Globalization.CultureInfo.InvariantCulture);
+                decimalPrice = _convertToEur(decimalPrice);
+                if (_isTooLong(30, name))
+                {
+                    itemName = WebUtility.UrlDecode(itemName);
+                    name = itemName;
+                }
+
+                var temp = Goal.CreateLocalGoal(name, decimalPrice);
+                return temp;
             }
-            catch (Exception ex)
+            catch(Exception ex)
             {
                 ExceptionHandler.Log(ex.ToString());
+                itemName = WebUtility.UrlDecode(itemName);
+                return Goal.CreateLocalGoal(itemName, 0);
```

Comment:
Later on maybe alerting the user that the parser has failed with a simple dialog box would be a nice feature

Reply:
So for example we could throw this further, and surround this parse statement with a try catch block in the front end and if we catch and exception we display the error to the user

Reply:
It's not that simple when post is called, because returning to the same page will move to the error page instead of alert, and the only simple solution I use to escape `POST` being called again on refresh is redirecting, so all exceptions will be lost :/

Reply:
Same problem will not allow to bubble exeptions to income,expenses,goals. And the alert will only be shown when data is not loaded from db, but not when adding/deleting etc. since it always redirects to this page and it "forgets" everything :/

Reply:
I see, well we will work with what we can, I don't think this will be too big of a problem for us, as long as we satisfy the requirement it's good